### PR TITLE
Update Header with application/json

### PR DESCRIPTION
**Pull request by safakkizkin**
#### Summary
Assigned content type application/json for NotFound (404), to stay true to the document.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/20878
JIRA: https://mattermost.atlassian.net/browse/MM-46605

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
@noxer - mind taking a look ? :)

File: web/web.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func Handle404(a app.AppIface, w http.ResponseWriter, r *http.Request) {
 	err := model.NewAppError("Handle404", "api.context.404.app_error", nil, "", http.StatusNotFound)
 	ipAddress := utils.GetIPAddress(r, a.Config().ServiceSettings.TrustedProxyIPHeader)
 	mlog.Debug("not found handler triggered", mlog.String("path", r.URL.Path), mlog.Int("code", 404), mlog.String("ip", ipAddress))
-
+	w.Header().Set("Content-Type", "application/json")
```
Comment (noxer) [inline web/web.go:64]: This should be moved into the `if IsAPICall(a, r) {` block. The other paths don't return JSON.

**Review by noxer — CHANGES_REQUESTED** [review]

**Comment by noxer** [comment]
Hi @safakkizkin and thank you for your contribution. I'd like to request a small change and then this should be good to go.

**Comment by safakkizkin** [comment]
Hi @noxer, yes you are totaly right!

**Review by noxer — APPROVED** [review]
Looks good, thank you @safakkizkin!

**Comment by noxer** [comment]
ping @isacikgoz

**Comment by noxer** [comment]
ping @isacikgoz

**Comment by safakkizkin** [comment]
rebased to be sure everything is great

**Review by noxer — APPROVED** [review]

**Review by isacikgoz — APPROVED** [review]

**Comment by noxer** [comment]
Merged, thank you!

**Pull request merged**